apps/web/src/lib/tools/discovery.ts

```diff
@@ -0,0 +1,90 @@
+/**
+ * Tool Discovery System
+ * Dynamically discovers available tools from their API routes
+ * This replaces the need for redundant tool config paths
+ */
+
+import { readdirSync, existsSync } from 'fs';
+import { join } from 'path';
+
+export interface ToolMetadata {
+  id: string;
+  name: string;
+  description: string;
+  version: string;
+  category: 'analysis' | 'research' | 'utility';
+  costEstimate?: string;
+  status: 'stable' | 'beta' | 'experimental';
+  apiPath: string;      // Derived: /api/tools/{id}
+  pagePath: string;     // Derived: /tools/{id}
+}
+
+/**
+ * Discovers all available tools by scanning the API routes directory
+ * and extracting metadata from the actual tool implementations
+ */
+export function discoverAvailableTools(): ToolMetadata[] {
+  const toolsApiDir = join(process.cwd(), 'src/app/api/tools');
```

Comment:
why do we do this? Why not import directly? This sync stuff seems weird. we should know where the files are and expect them.